### ISSUE TITLE
eqJoin between server tables with `uuid` identifierFormat leads to guarantee failure

### ISSUE THREAD

**Issue by mbroadst** [rethinkdb/rethinkdb#6108]
Tested inside the data explorer:

**reql query:**
```
r.db('rethinkdb').table('table_status', { identifierFormat: 'uuid' })
  .eqJoin('raft_leader', r.db('rethinkdb').table('server_status'));
```

**result:**
```
e: SANITY CHECK FAILED: [false] at `./src/rdb_protocol/context.hpp:155` (server is buggy).  Backtrace:
Wed Sep 14 20:07:42 2016

1 [0xab45d0]: backtrace_t::backtrace_t() at 0xab45d0 (/usr/bin/rethinkdb)
2 [0xab489e]: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at 0xab489e (/usr/bin/rethinkdb)
3 [0x93fe33]: ql::runtime_sanity_check_failed(char const*, int, char const*, std::string const&) at 0x93fe33 (/usr/bin/rethinkdb)
4 [0x8aaabb]: base_table_t::read_all_with_sindexes(ql::env_t*, std::string const&, ql::backtrace_id_t, std::string const&, ql::datumspec_t const&, sorting_t, read_mode_t) at 0x8aaabb (/usr/bin/rethinkdb)
5 [0x95327d]: ql::table_t::get_all_with_sindexes(ql::env_t*, ql::datumspec_t const&, std::string const&, ql::backtrace_id_t) at 0x95327d (/usr/bin/rethinkdb)
6 [0xa0c33e]: ql::eq_join_datum_stream_t::next_raw_batch(ql::env_t*, ql::batchspec_t const&) at 0xa0c33e (/usr/bin/rethinkdb)
7 [0xa09b7a]: ql::eager_datum_stream_t::next_grouped_batch(ql::env_t*, ql::batchspec_t const&, std::map<ql::datum_t, std::vector<ql::datum_t, std::allocator<ql::datum_t> >, optional_datum_less_t, std::allocator<std::pair<ql::datum_t const, std::vector<ql::datum_t, std::allocator<ql::datum_t> > > > >*) at 0xa09b7a (/usr/bin/rethinkdb)
8 [0xa0a255]: ql::eager_datum_stream_t::next_batch_impl(ql::env_t*, ql::batchspec_t const&) at 0xa0a255 (/usr/bin/rethinkdb)
9 [0xa05547]: ql::datum_stream_t::next_batch(ql::env_t*, ql::batchspec_t const&) at 0xa05547 (/usr/bin/rethinkdb)
10 [0x8c896f]: ql::query_cache_t::ref_t::serve(ql::env_t*, ql::response_t*) at 0x8c896f (/usr/bin/rethinkdb)
11 [0x8caa87]: ql::query_cache_t::ref_t::fill_response(ql::response_t*) at 0x8caa87 (/usr/bin/rethinkdb)
12 [0x951df3]: rdb_query_server_t::run_query(ql::query_params_t*, ql::response_t*, signal_t*) at 0x951df3 (/usr/bin/rethinkdb)
13 [0x7da0b3]: query_server_t::handle(http_req_t const&, http_res_t*, signal_t*) at 0x7da0b3 (/usr/bin/rethinkdb)
14 [0xa9cabd]: routing_http_app_t::handle(http_req_t const&, http_res_t*, signal_t*) at 0xa9cabd (/usr/bin/rethinkdb)
15 [0xa9cabd]: routing_http_app_t::handle(http_req_t const&, http_res_t*, signal_t*) at 0xa9cabd (/usr/bin/rethinkdb)
16 [0xa95357]: http_server_t::handle_conn(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t) at 0xa95357 (/usr/bin/rethinkdb)
17 [0xa98b34]: std::_Function_handler<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&), boost::_bi::bind_t<void, boost::_mfi::mf2<void, http_server_t, scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t>, boost::_bi::list3<boost::_bi::value<http_server_t*>, boost::arg<1>, boost::_bi::value<auto_drainer_t::lock_t> > > >::_M_invoke(std::_Any_data const&, scoped_ptr_t<linux_tcp_conn_descriptor_t>&) at 0xa98b34 (/usr/bin/rethinkdb)
18 [0x7a407f]: linux_nonthrowing_tcp_listener_t::handle(int) at 0x7a407f (/usr/bin/rethinkdb)
19 [0x7bf297]: coro_t::run() at 0x7bf297 (/usr/bin/rethinkdb)
 in:
r.db("rethinkdb").table("table_status", {"identifierFormat": "uuid"}).eqJoin("raft_leader", r.db("rethinkdb").table("server_status"))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by nighelles** [comment]
This is fixed in next.

**Comment by AtnNn** [comment]
A fix for this issue has been released in [RethinkDB 2.3.6](https://rethinkdb.com/blog/2.3.6-release/)
